plugins/app/src/main/java/com/mapbox/mapboxsdk/plugins/testapp/activity/GeoJsonActivity.java

```diff
@@ -0,0 +1,250 @@
+    private static String TAG = "GeoJsonActivity";
+    @BindView(R.id.mapView)
+    MapView mapView;
+
+    @BindView(R.id.fabURL)
+    FloatingActionButton urlFab;
+
+    @BindView(R.id.fabAssets)
+    FloatingActionButton assetsFab;
+
+    @BindView(R.id.progressBar)
+    ProgressBar progressBar;
+
+    private MapboxMap mapboxMap;
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_geojson);
+        ButterKnife.bind(this);
+        progressBar.getIndeterminateDrawable().setColorFilter(Color.RED,
+                android.graphics.PorterDuff.Mode.SRC_IN);
+        mapView.setStyleUrl(Style.MAPBOX_STREETS);
+        mapView.onCreate(savedInstanceState);
+        mapView.getMapAsync(this);
+    }
+
+    @Override
+    public void onMapReady(MapboxMap mapboxMap) {
+        this.mapboxMap = mapboxMap;
+        mapboxMap.moveCamera(CameraUpdateFactory.newLatLngZoom(new LatLng(32.6546, 51.6680), 7));
+
+    }
+
+    @OnClick(R.id.fabURL)
+    public void onURLFabClick() {
+        mapboxMap.clear();
+        if (mapboxMap != null) {
+            new GeoJsonPluginBuilder()
+                    .withContext(this)
+                    .withMap(mapboxMap)
+                    .withUrl("https://earthquake.usgs.gov/earthquakes/feed/v1.0/summary/all_day.geojson")
+                    .withOnLoadingURL(new OnLoadingGeoJsonListener() {
+                        @Override
+                        public void onPreLoading() {
+                            progressBar.setVisibility(View.VISIBLE);
+                        }
+
+                        @Override
+                        public void onLoaded() {
+                            Toast.makeText(GeoJsonActivity.this, "GeoJson data from url loaded", Toast.LENGTH_LONG).show();
+                            progressBar.setVisibility(View.INVISIBLE);
+                        }
+
+                        @Override
+                        public void onLoadFailed(Exception e) {
+                            progressBar.setVisibility(View.INVISIBLE);
+                            Toast.makeText(GeoJsonActivity.this, "Error occur during load GeoJson data from url. see logcat", Toast.LENGTH_LONG).show();
+                            e.printStackTrace();
+                        }
+                    })
+                    .withMarkerClickListener(new OnMarkerEventListener() {
+                        @Override
+                        public void onMarkerClickListener(Marker marker, JSONObject properties) {
+                            try {
+                                Toast.makeText(GeoJsonActivity.this, properties.getString("title"), Toast.LENGTH_LONG).show();
+                            } catch (JSONException e) {
+                                e.printStackTrace();
+                            }
+                        }
+                    })
+                    .draw();
```

Comment:
a builder pattern normally ends with calling a method with using the `build()` method signature